src/dql_parser.yrl

```diff
@@ -185,18 +185,30 @@ maybe_group_by -> mfrom : '$1'.
 grouping -> maybe_scoped_dvar : ['$1'].
 grouping -> maybe_scoped_dvar ',' grouping : ['$1'] ++ '$3'.
 
+mfrom -> kw_all kw_from bucket : #{
+                          op => lookup,
+                          args => ['$3', undefined],
+                          return => metric
+                        }.
+
+mfrom -> kw_all kw_from bucket kw_where where :#{
+                                          op => lookup,
+                                          args => ['$3', undefined, '$5'],
+                                          return => metric
+                                         }.
+
```

Comment:
Nice suggestion, I started off creating a new operation entirely, called `lookup_all`.  Then I simplified to above, and using a new NT would make it even better.